File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.swift
import Foundation

func solution(_ arr:[Int], _ query:[Int]) -> [Int] {
    
    var result = arr
    
    for index in 0..<query.count {
        if index.isMultiple(of: 2) {
            result.removeLast(result.count - query[index] - 1)
        } else {
            result.removeFirst(query[index])
        }
    }
    
    return result
}

File: 프로그래머스/0/181893. 배열 조각하기/배열 조각하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<int> query) {

    for (int i = 0; i < query.size(); i++) {
        if (i % 2 == 0) {
            int size = query[i] + 1;
            vector<int> temp(arr.begin(), arr.begin() + size);
            arr = temp;
        } else {
            int size = query[i];
            vector<int> temp(arr.begin() + size, arr.end());
            arr = temp;
        }
    }

    return arr;
}